### PR TITLE
API doc for U2F factor enroll and challenge in authentication API

### PR DESCRIPTION
**Pull request by srinivasanagandla-okta**
- *Added section for enroll and verify API details for U2F factor*
- *Also fixed references to rain.okta1.com and changed them to your-domain.okta.com*

Please review @mystiberry-okta @tthompson-okta

### CODE REVIEW

**Comment by mystiberry-okta** [comment]
Looks great to me. Ready to merge (or should it wait for next week?) @srinivasanagandla-okta I did a few punctuation and capitalization edits, nothing major.

**Comment by srinivasanagandla-okta** [comment]
@mystiberry-okta I have tagged @tthompson-okta as well. I will wait for comments from him.

**Comment by srinivasanagandla-okta** [comment]
@mystiberry-okta This feature is still in EA. I am wondering if I need to mention that anywhere in the documentation.

**Comment by mystiberry-okta** [comment]
I added the tag for EA. Let me know if it needs anything else.

File: _source/_docs/api/resources/authn.md [hunk 1]
```diff
@@ -1656,6 +1657,121 @@ In this example we just enrolled and activated Duo but the question and SMS fact
...
+        "allow": [
+          "POST"
+        ]
+      }
+    }
+  }
+}
+~~~
+
+##### Response Example: Get Registration Information by Calling the U2F Javascript Call
```
Comment (karlmcguinness-okta) [inline _source/_docs/api/resources/authn.md]: I think this step is more related to activation then enroll as you need the output of device registration to complete activation

#### Enroll U2F Factor
#### Activate U2F Factor
#####  Web U2F Device Registration

Comment (karlmcguinness-okta) [inline _source/_docs/api/resources/authn.md]: I would also indicate that this is sample and not part of Okta API.  The browser needs to support U2F APIs.  Might be nice to provide a browser compatibility matrix.

Note it is also possible to use a non-web flow like https://developers.yubico.com/pam-u2f/

Comment (srinivasanagandla-okta) [inline _source/_docs/api/resources/authn.md]: @karlmcguinness-okta Thanks for the feedback. I will move the javascript sample code to activation section.

Comment (srinivasanagandla-okta) [inline _source/_docs/api/resources/authn.md]: Regarding using it non webflow, currently appId is fixed to https://domain.okta.com. I think that would create issues when its being used in non-web flow. 

Once we support configurable appId then we can talk about it in using this in non-web flow.

**Pull request merged**